Sources/Classes/RSEventRepository.m

```diff
@@ -481,30 +485,49 @@ - (void) __checkApplicationUpdateStatus {
                               UIApplicationDidBecomeActiveNotification ]) {
         [nc addObserver:self selector:@selector(handleAppStateNotification:) name:name object:UIApplication.sharedApplication];
     }
+#else
+    for (NSString *name in @[ WKApplicationDidEnterBackgroundNotification,
+                              WKApplicationDidFinishLaunchingNotification,
+                              WKApplicationWillEnterForegroundNotification,
+                              WKApplicationWillResignActiveNotification,
+                              WKApplicationDidBecomeActiveNotification ]) {
+        [nc addObserver:self selector:@selector(handleAppStateNotification:) name:name object:nil];
+    }
+#endif
 }
 
 - (void) handleAppStateNotification: (NSNotification*) notification {
+#if !TARGET_OS_WATCH
     if ([notification.name isEqualToString:UIApplicationDidFinishLaunchingNotification]) {
         [self _applicationDidFinishLaunchingWithOptions:notification.userInfo];
     } else if ([notification.name isEqualToString:UIApplicationWillEnterForegroundNotification]) {
         [self _applicationWillEnterForeground];
     } else if ([notification.name isEqualToString: UIApplicationDidEnterBackgroundNotification]) {
         [self _applicationDidEnterBackground];
     }
+#else
+    if ([notification.name isEqualToString:WKApplicationDidFinishLaunchingNotification]) {
+        [self _applicationDidFinishLaunchingWithOptions:notification.userInfo];
+    } else if ([notification.name isEqualToString: WKApplicationDidBecomeActiveNotification]) {
+        [self _applicationWillEnterForeground];
+    } else if ([notification.name isEqualToString: WKApplicationWillResignActiveNotification]) {
+        [self _applicationDidEnterBackground];
+    }
+#endif
 }
 
 - (void)_applicationDidFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
     if (!self->config.trackLifecycleEvents) {
         return;
     }
     NSString *previousVersion = [preferenceManager getBuildVersionCode];
-    
     NSString *currentVersion = [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"];
     
     if (!previousVersion) {
         [[RSClient sharedInstance] track:@"Application Installed" properties:@{
             @"version": currentVersion
         }];
+        [preferenceManager saveBuildVersionCode:currentVersion];
```

Comment:
I think previous implementation is correct.

Reply:
Actually, we need to save the version of the app in the user defaults, either when the app is installed or when the app got updated, as it is created or changed in those two scenarios itself. 

But in the previous implementation, the only problem is we are overriding the version in the user defaults every time the app is opened even though it is not needed. 

So I had changed it to this way to decrease the unnecessary writes to the user defaults.

Does it make sense (or) shall we move to the previous implementation?

Reply:
Okay Understand.